test/src/e2e.dynval/dv.n-1.test.ts

```diff
@@ -0,0 +1,184 @@
+
+import * as chai from "chai";
+import { expect } from "chai";
+import * as chaiAsPromised from "chai-as-promised";
+import { SDK } from "codechain-sdk";
+import * as stake from "codechain-stakeholder-sdk";
+import "mocha";
+
+import { validators as originalValidators } from "../../tendermint.dynval/constants";
+import { faucetAddress, faucetSecret } from "../helper/constants";
+import { PromiseExpect } from "../helper/promise";
+import { withNodes } from "./setup";
+import CodeChain from "../helper/spawn";
+
+async function getPossibleAuthors(
+    sdk: SDK,
+    blockNumber: number
+): Promise<null | string[]> {
+    return new Promise((resolve, reject) => {
+        sdk.rpc
+            .sendRpcRequest("chain_getPossibleAuthors", [blockNumber])
+            .then(result => {
+                if (
+                    result === null ||
+                    result.every((str: any) => typeof str === "string")
+                ) {
+                    resolve(result);
+                }
+                return reject(
+                    Error(
+                        `Expected chain_getPossibleAuthors to return an array of string or null but it returned ${result}`
+                    )
+                );
+            })
+            .catch(reject);
+    });
+}
+
+chai.use(chaiAsPromised);
+
+const [alice, , , ...otherDynValidators] = originalValidators;
+const allDynValidators = [alice, ...otherDynValidators];
+
+describe("Dynamic Validator N -> N-1", function() {
+    this.timeout(0);
+
+    const promiseExpect = new PromiseExpect();
+    const termSeconds = 20;
+
+    const nodes = withNodes(this, {
+        promiseExpect,
+        validators: allDynValidators.map((signer, index) => ({
+            signer,
+            delegation: 5_000,
+            deposit: 10_000_000 - index // tie-breaker
+        })),
+        overrideParams: {
+            termSeconds,
+            maxNumOfValidators: 8
+        }
+    });
+
+    describe("A node is imprisoned to jail", async function() {
+        it("alice should be dropped out from validator list", async function() {
+            const margin = 1.2;
+            const aliceNode = nodes[0];
+            const checkingNode = nodes[1];
+
+            await aliceNode.clean();
+
+            // not to make timing issue, wait until the term 3.
+            await checkingNode.waitForTermChange(3, termSeconds * margin * 2);
+
+            const blockNumber = await checkingNode.sdk.rpc.chain.getBestBlockNumber();
+            const termMedata = await stake.getTermMetadata(
+                checkingNode.sdk,
+                blockNumber
+            );
+            const currentTermInitialBlockNumber =
+                termMedata!.lastTermFinishedBlockNumber + 1;
+            const validatorsAfter = await getPossibleAuthors(
+                checkingNode.sdk,
+                currentTermInitialBlockNumber
+            );
+
+            expect(validatorsAfter).contains.all.members(
```

Comment:
Please check if Alice is missing.
Please check that the number of the validator list has decreased.

Reply:
I added both checks